lib/services/android-project-service.ts

```diff
@@ -826,8 +903,39 @@ export class AndroidProjectService extends projectServiceBaseLib.PlatformProject
 		await adb.executeShellCommand(["rm", "-rf", deviceRootPath]);
 	}
 
-	public async checkForChanges(): Promise<void> {
-		// Nothing android specific to check yet.
+	public async checkForChanges(
+		changesInfo: IProjectChangesInfo,
+		prepareData: AndroidPrepareData,
+		projectData: IProjectData
+	): Promise<void> {
+		//we need to check for abi change in connected device vs last built
+		const deviceDescriptors = this.$liveSyncProcessDataService.getDeviceDescriptors(
+			projectData.projectDir
+		);
+		const platformData = this.getPlatformData(projectData);
+		deviceDescriptors.forEach(deviceDescriptor=>{
+			const buildData = deviceDescriptor.buildData as AndroidBuildData;
+			if (buildData.buildFilterDevicesArch) {
+				const outputPath = platformData.getBuildOutputPath(deviceDescriptor.buildData);
+				const apkOutputPath = path.join(outputPath, prepareData.release ? "release" : "debug");
+				if (!this.$fs.exists(apkOutputPath)) {
+					return;
+				}
+				// check if we already build this arch
+				// if not we need to say native has changed
+				const device = this.$devicesService.getDevicesForPlatform(deviceDescriptor.buildData.platform).filter(d=>d.deviceInfo.identifier === deviceDescriptor.identifier)[0];
+				const abis = device.deviceInfo.abis.filter(a=>!!a && a.length)[0];
+				
+				const directoryContent = this.$fs.readDirectory(apkOutputPath);
+				const regexp = new RegExp(`${abis}.*\.apk`);
+				const files = _.filter(directoryContent, (entry: string) => {
+					return regexp.test(entry);
+				});
+				if (files.length === 0) {
+					changesInfo.nativeChanged = true;
```

Comment:
@farfromrefug this seems to be causing a rebuild every time you run `ns debug android`.

Reply:
@farfromrefug possible to adjust that portion to not rebuild on initial run?

Reply:
@NathanWalker we started discussing that with @rigor789. I proposed to add a test to detect universal builds (through apk name). If we agree on that (or another solution) i ll implement it